tests/unit/s2n_client_hello_recv_test.c

```diff
@@ -375,6 +376,48 @@ int main(int argc, char **argv)
         s2n_connection_free(server_conn);
     }
 
+    /* Test that curve selection will be NIST P-256 when tls12 client does not sending curve extension. */
+    {
+        S2N_BLOB_FROM_HEX(tls12_client_hello_no_curves,
+            "030307de81928fe1" "7cba77904c2798da" "2521a76b013a16e4" "21ade32208f658d4" "327d000048000400"
+            "05000a0016002f00" "3300350039003c00" "3d0067006b009c00" "9d009e009fc009c0" "0ac011c012c013c0"
+            "14c023c024c027c0" "28c02bc02cc02fc0" "30cca8cca9ccaaff" "04ff0800ff010000" "30000d0016001404"
+            "0105010601030104" "0305030603030302" "010203000b000201" "00fe01000c000a00" "17000d0013000100"
+            "0a");
+
+        /* The above code is generated the following code,
+            disabling s2n_client_supported_groups_extension
+            from client_hello_extensions (s2n_extension_type_lists.c)
+            and exporting the resulting client_conn->handshake.io.blob
+
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, tls12_config));
+        EXPECT_SUCCESS(s2n_client_hello_send(client_conn));
+        EXPECT_EQUAL(client_conn->client_protocol_version, S2N_TLS12);
+        EXPECT_EQUAL(client_conn->client_hello_version, S2N_TLS12);
```

Comment:
Why store this as a blob instead of just executing it as part of the test? What's the benefit?

Reply:
it is not possible to execute the current test with code unless we can expose not sending supported_groups extension